### PR TITLE
Remove TableLayout and related objects

### PR DESCRIPTION
**Pull request by Praveen2112**
Closes https://github.com/trinodb/trino/issues/781

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/split/SplitManager.java [hunk 1]
```diff
@@ -81,25 +72,13 @@ public SplitSource getSplits(
...
-                            .get()
-                            .getNewTableHandle()
-                            .getLayout().get());
-
-            source = splitManager.getSplits(table.getTransaction(), connectorSession, layout, splitSchedulingStrategy);
-        }
-        else {
-            source = splitManager.getSplits(
+        ConnectorSplitSource source = splitManager.getSplits(
                     table.getTransaction(),
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/split/SplitManager.java:76]: I think this indentation should change since it's no longer in a block

**Review by electrum — APPROVED** [review]
This is great!

**Review by kokosing — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java [hunk 2]
```diff
@@ -332,4 +308,54 @@ static Expression createResultingPredicate(
 
         return expression;
     }
+
+    public static TupleDomain<ColumnHandle> computeEnforced(TupleDomain<ColumnHandle> predicate, TupleDomain<ColumnHandle> unenforced)
+    {
+        if (predicate.isNone()) {
+            // If the engine requests that the connector provides a layout with a domain of "none". The connector can have two possible reactions, either:
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java:315]: Still some references to "layout"

**Comment by Praveen2112** [comment]
@martint AC

**Review by martint — APPROVED** [review]

**Pull request merged**